int32 id
int32 size
int32[] start
int32[] end
int32[] data
